Add markdownz to editable fields

simoneduca opened the pull request:
List of fields that need markdown support

- [x] all project pages
- [x] tutorial (step only)
- [x] Field guide (content only)

eatyourgreens commented on src/containers/ProjectContentsContainer.jsx:
```diff
@@ -113,8 +113,15 @@ class ProjectContentsContainer extends Component {
-            <h2>Translation</h2>
-            <input type="text" onChange={this.handleChange} autoFocus placeholder="Translate some text" />
+            <h2>Your translation</h2>
+            <MarkdownEditor
```
This should be a textarea or text input if the field is plain text.

You should add a flag to fields that need the markdown editor, eg. `data-markdown=true` then check the flag here (`event.target.getAttribute('data-markdown')` and render either a plain text input or the markdown editor.

simoneduca replied:
Agh, my bad sorry.

eatyourgreens commented on src/styles/components/markdown-editor-preview.styl:
```diff
@@ -0,0 +1,2 @@
+.markdown-editor-preview
+  display none
```
Do you think people will want to use the preview button to look at their changes? We can leave that until later but worth keeping in mind.

simoneduca replied:
Having a preview would be nice I think. But I'm not sure it works properly. The property `previewing` is false by the default, so I don't understand why it was showing anyway, without this css.

eatyourgreens replied:
I'm happy to leave this and figure it out later.

At worst, we could leave the preview showing under the editor, but only if it updates when you change the markdown.

simoneduca commented:
Should be fixed now. Also I realised the markdown viewer I added to field guide, tutorials and project pages wasn't in the right place so I removed it.

eatyourgreens commented:
👍🏻 A couple of small comments. Nothing major.
